src/bridges/chainlink-ccip/CcipGhoBridge.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {Ownable} from 'solidity-utils/contracts/oz-common/Ownable.sol';
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {LinkTokenInterface} from '@chainlink/contracts/src/v0.8/shared/interfaces/LinkTokenInterface.sol';
+import {CCIPReceiver} from '@chainlink/contracts-ccip/src/v0.8/ccip/applications/CCIPReceiver.sol';
+import {IRouterClient} from '@chainlink/contracts-ccip/src/v0.8/ccip/interfaces/IRouterClient.sol';
+import {Client} from '@chainlink/contracts-ccip/src/v0.8/ccip/libraries/Client.sol';
+
+import {ICcipGhoBridge} from './ICcipGhoBridge.sol';
+
+/**
+ * @title CcipGhoBridge
+ * @author LucasWongC
+ * @notice Helper contract to bridge GHO using Chainlink CCIP
+ */
+contract CcipGhoBridge is ICcipGhoBridge, CCIPReceiver, Ownable {
+  /// @dev Chainlink CCIP router address
+  address public immutable ROUTER;
+  /// @dev LINK token address
+  address public immutable LINK;
+  /// @dev GHO token address
+  address public immutable GHO;
+
+  /// @dev Address of bridge (chainSelector => bridge address)
+  mapping(uint64 => address) public bridges;
+
+  /**
+   * @param _router The address of the Chainlink CCIP router
+   * @param _link The address of the LINK token
+   * @param _gho The address of the GHO token
+   * @param _owner The address of the contract owner
+   */
+  constructor(address _router, address _link, address _gho, address _owner) CCIPReceiver(_router) {
+    ROUTER = _router;
+    LINK = _link;
+    GHO = _gho;
+
+    _transferOwnership(_owner);
+  }
+
+  receive() external payable {}
+
+  /// @dev Checks if the destination bridge has been set up
+  modifier checkDestination(uint64 chainSelector) {
+    if (bridges[chainSelector] == address(0)) {
+      revert UnsupportedChain();
+    }
+    _;
+  }
+
+  /// @inheritdoc ICcipGhoBridge
+  function transfer(
+    uint64 destinationChainSelector,
+    Transfer[] calldata transfers,
+    PayFeesIn payFeesIn
+  ) external payable checkDestination(destinationChainSelector) returns (bytes32 messageId) {
```

Comment:
This function is callable by anyone, we should only allow it to be called by OwnerOrGuardian and add a guaridan role to it